src/main/scala/is/hail/methods/FilterIntervals.scala

```diff
@@ -12,13 +12,13 @@ object FilterIntervals {
   def apply(vsm: MatrixTable, intervals: java.util.ArrayList[Interval], keep: Boolean): MatrixTable = {
     vsm.requireRowKeyVariant("filter_intervals")
     val locusField = vsm.rowType.fieldByName(vsm.rowKey(0))
-    val iList = IntervalTree(locusField.typ.ordering, intervals.asScala.toArray)
+    val iList = IntervalTree[Any](locusField.typ.ordering, intervals.asScala.toArray)
```

Comment:
I don't see why the type parameter needs to get added to IntervalTree.  It doesn't seem to be used profitably anywhere.  Did you try taking it out again?  What goes wrong?